source/common/uma_pools/disjoint_pool.hpp

```diff
@@ -0,0 +1,83 @@
+//===---------- disjoint_pool.hpp - Allocator for USM memory --------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef USM_ALLOCATOR
+#define USM_ALLOCATOR
+
+#include <atomic>
+#include <memory>
+
+#include "../uma_helpers.hpp"
+
+namespace usm {
+
+// Configuration for specific USM allocator instance
+class DisjointPoolConfig {
```

Comment:
might be better to move this to `disjoint_pool_config.hpp`

Reply:
I'm not sure. I wanted to keep the disjoint_pool implementation self-contained (which also means this configuration) but independent on how many pools we create.

disjoint_pool_config only provides defaults and parses string for a particular use-case (USM allocations with separate pools per device, host, shared, sharedReadOnly). I think I should name it better tough - perhaps disjoint_pool_config_parser or something.